### PR TITLE
refactor(ui): Scoped CSS Reset

### PR DESCRIPTION
**Pull request by joe-bell**
## Description

Adds scoped (and unscoped global styling for variables) to ensure elements are unaffected from consumer global styles

_Implementation details provided via inline comments_

### Demo

Without consumer base styles

CleanShot 2024-09-03 at 15 00 56@2x

Without any consumer base styles

CleanShot 2024-09-03 at 15 00 36@2x

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/ui/src/tailwind.config.ts [hunk 1]
```diff
@@ -111,6 +114,236 @@ const config = {
...
+              ?.map(item => item.trim())
+              .filter(item => item !== '')
+              .map(selector => `[class^='cl-'] ${selector}, [class*=' cl-'] ${selector}`)
+              .join(', ')})`]: properties,
+          }),
+          {} as Record<string, string | Record<string, string>>,
+        );
+
+        addBase(scopedStyles);
+      };
```
Comment (alexcarpenter) [inline packages/ui/src/tailwind.config.ts:148]: Note from discussion: ensure this is not just targeting child selectors.

File: packages/ui/src/tailwind.config.ts [hunk 2]
```diff
@@ -111,6 +114,236 @@ const config = {
...
+      addBase({
+        ':where(:root)': {
+          '--cl-font-family':
+            '-apple-system, BlinkMacSystemFont, avenir next, avenir, segoe ui, helvetica neue, helvetica,\n    Cantarell, Ubuntu, roboto, noto, arial, sans-serif',
+          '--cl-color-danger': '0 84% 60%',
+          '--cl-color-success': '142 71% 45%',
+          '--cl-color-warning': '25 95% 53%',
+          '--cl-radius': '0.375rem',
+          '--cl-spacing-unit': '1rem',
+          '--cl-font-size': '0.8125rem',
```
Comment (alexcarpenter) [inline packages/ui/src/tailwind.config.ts:320]: Note for future PR to add all of the default color variables here.

**Review by alexcarpenter — APPROVED** [review]
Looks good, added a small commit ensuring some elements have a background defined to avoid global styles leaking in https://github.com/clerk/javascript/pull/4075/commits/e2372c7be0fe9107ff3b4a8a0cdb44b71a4f5711

File: packages/ui/src/tailwind.config.ts (hunk 2 above)
Comment (joe-bell) [reply packages/ui/src/tailwind.config.ts:327]: Following up

**Pull request merged**